src/citationnet/static/force-citationNet.js

```diff
@@ -0,0 +1,419 @@
+        if (this.distanceFromInputNode) {
+            nodes.forEach((node) => {
+                if (node.attributes.nodeyear >= this.inputNode.attributes.nodeyear) {
+                    node.fz = 5 * (node.attributes.nodeyear - this.inputNode.attributes.nodeyear);
+                } else {
+                    node.fz = 5 * (node.attributes.nodeyear - this.inputNode.attributes.nodeyear);
+                }
+            });
+            this.inputNode.fz = 0;
+            document.getElementById("btnDistanceFromInputNode").style.fontWeight = "normal";
+            this.distanceFromInputNode = false;
+        } else {
+            nodes.forEach((node) => {
+                if (node.attributes.nodeyear >= this.inputNode.attributes.nodeyear) {
+                    node.fz = 5 * (node.attributes.nodeyear - this.inputNode.attributes.nodeyear + 20);
+                } else {
+                    node.fz = 5 * (node.attributes.nodeyear - this.inputNode.attributes.nodeyear - 20);
+                }
+            });
+            this.inputNode.fz = 0;
+            document.getElementById("btnDistanceFromInputNode").style.fontWeight = "bold";
+            this.distanceFromInputNode = true;
+        }
+        console.log("reheating");
+        this.graph.d3ReheatSimulation();
+    }
+
+    /**
+     * Get field of research statistics
+     * @returns {Array}
+     */
+    async getStats() {
+        this.data = await fetchJSON(this.jsondata);
+        this.processData();
+
+        var nodes = this.data.nodes;
+        // var nodes = this.graph.graphData().nodes
+        this.stats = [];
+        // var cumulative = 0.0;
+
+        for (var division in fieldOfResearchDivisions){
+            const reducer = (accumulator, curr) => accumulator + curr;
+            var divnodesContain = nodes.filter(node => node.attributes.category_for.includes(division + ':'));
+            // console.log(divnodesContain)
+            var divnodesElements = divnodesContain.map(node => node.attributes.category_for.split(';')).flat(1);
+            // console.log(divnodesElements)
+            var divnodes = divnodesElements.filter(forcode => forcode.includes(division));
+            // console.log(divnodes)
+            var divnumVals = divnodes.map(elem => elem.split(':')[1]);
+            // console.log(divnumVals)
+            // const myArray = text.split(";").filter(forcode => forcode.includes('03')).map(forcode => forcode.split(':')[1]).map(val => parseFloat(val)).reduce(reducer, 0);
+            var x = divnumVals.map(val => parseFloat(val)).reduce(reducer, 0)/nodes.length;
+            // console.log(division, x)
+            this.stats.push({ 'category': fieldOfResearchDivisions[division][0], 'color': fieldOfResearchDivisions[division][1], 'value': x, 'amount': divnodesContain.length });
+            // cumulative += x
+        }
+
+        return this.stats;
+    }
+}
+
+
+
+const lutFOR = new Lut.Lut('rainbow', 22);
```

Comment:
Not a big deal, but I would put constants like this at the top of the file or in a separate file.